clients/drcachesim/tools/opcode_mix.cpp

```diff
@@ -81,30 +81,35 @@ opcode_mix_t::opcode_mix_t(const std::string &module_file_path, unsigned int ver
 {
 }
 
+void
+opcode_mix_t::make_decode_cache(shard_data_t *shard, void *dcontext)
+{
+    shard->decode_cache = std::unique_ptr<decode_cache_t<opcode_data_t>>(
+        new decode_cache_t<opcode_data_t>(dcontext,
+                                          /*persist_decoded_instrs=*/false));
+}
+
+std::string
+opcode_mix_t::init_decode_cache(shard_data_t *shard, void *dcontext,
+                                offline_file_type_t filetype)
+{
+    make_decode_cache(shard, dcontext);
+    std::string err;
+    if (!TESTANY(OFFLINE_FILE_TYPE_ENCODINGS, filetype)) {
```

Comment:
I think the new library has to document that currently users have to set the ISA mode to regdeps.  Could we just have the library do it now in its init method?  I'm just afraid of people writing analyzers for the new public traces and they all fail to decode when they try to use this library out of the box.

Reply:
>  Could we just have the library do it now in its init method?

Done